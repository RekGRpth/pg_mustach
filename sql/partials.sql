\unset ECHO
\set QUIET 1
\pset format unaligned
\pset tuples_only true
\pset pager off
\set ON_ERROR_ROLLBACK 1
\set ON_ERROR_STOP true
BEGIN;
CREATE EXTENSION pg_mustach;
COPY (
SELECT 1, 'Basic Behavior', mustach('{"text": "from partial"}', E'\"{{>text}}\"')
UNION
SELECT 2, 'Failed Lookup', mustach('{}', E'\"{{>text}}\"')
UNION
SELECT 3, 'Context', mustach('{"text": "content", "partial": "*{{text}}*"}', E'\"{{>partial}}\"')
UNION
SELECT 4, 'Recursion', mustach('{"content": "X", "nodes": [{"content": "Y", "nodes": []}], "node": "{{content}}<{{#nodes}}{{>node}}{{/nodes}}>"}', E'{{>node}}')
UNION
SELECT 5, 'Surrounding Whitespace', mustach('{"partial": "\t|\t"}', E'| {{>partial}} |')
UNION
SELECT 6, 'Inline Indentation', mustach('{"data": "|", "partial": ">\n>"}', E'  {{data}}  {{> partial}}\n')
UNION
SELECT 7, 'Standalone Line Endings', mustach('{"partial": ">"}', E'|\r\n{{>partial}}\r\n|')
UNION
SELECT 8, 'Standalone Without Previous Line', mustach('{"partial": ">\n>"}', E'  {{>partial}}\n>')
UNION
SELECT 9, 'Standalone Without Newline', mustach('{"partial": ">\n>"}', E'>\n  {{>partial}}')
UNION
SELECT 10, 'Standalone Indentation', mustach('{"content": "<\n->", "partial": "|\n{{{content}}}\n|\n"}', E'\\\n {{>partial}}\n/\n')
UNION
SELECT 11, 'Padding Whitespace', mustach('{"boolean": true, "partial": "[]"}', E'|{{> partial }}|')
ORDER BY 1
) TO stdout;
ROLLBACK;
