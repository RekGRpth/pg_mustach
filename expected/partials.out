\unset ECHO
1	Basic Behavior	"from partial"
2	Failed Lookup	""
3	Context	"*content*"
4	Recursion	X<Y<>>
5	Surrounding Whitespace	| \t|\t |
6	Inline Indentation	  |  >\n>\n
7	Standalone Line Endings	|\r\n>|
8	Standalone Without Previous Line	  >\n  >>
9	Standalone Without Newline	>\n  >\n  >
10	Standalone Indentation	\\\n |\n <\n->\n |\n/\n
11	Padding Whitespace	|[]|
